nexus/src/app/background/tasks/region_replacement_driver.rs

```diff
@@ -119,36 +120,31 @@ impl RegionReplacementDriver {
                 // (or determine if it is complete).
 
                 let request_id = request.id;
-
-                let result = self
-                    .saga_request
-                    .send(sagas::SagaRequest::RegionReplacementDrive {
-                        params: sagas::region_replacement_drive::Params {
-                            serialized_authn:
-                                authn::saga::Serialized::for_opctx(opctx),
-                            request,
-                        },
-                    })
-                    .await;
-
+                let params = sagas::region_replacement_drive::Params {
+                    serialized_authn: authn::saga::Serialized::for_opctx(opctx),
+                    request,
+                };
+                let result = futures::future::ready(
+                    SagaRegionReplacementDrive::prepare(&params),
+                )
+                .and_then(|saga_dag| self.sagas.saga_start(saga_dag))
+                .await;
```

Comment:
this felt a bit strange to me at first glance, but then i see it's so that we can use `and_then` to handle the errors from both functions in one place...

I don't really think this needs to be changed, and I'm not sure if there's a way to make it clearer, just calling out that it took me few seconds.

Reply:
I totally agree and it's one of the things I wasn't super happy about but also didn't have a better idea.  (I find this a lot when trying to chain the usual result/option combinators with an async operation in the middle.)

Reply:
Yeah, the other approach that occurs to me would be to wrap all this in an `async fn` where you can just `?` it, but I don't think that's really worth the effort for basically two lines of code. I think what you've got here is fine, though.

Reply:
actually --- and i had to go and [double-check this in the playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0a449febc7423f833be4d497728c8b0d) --- it turns out that an `async` block is a scope that catches `?`s, so you _could_ alternatively write this, instead:
```suggestion
                let result = async {
                    let saga_dag = SagaRegionReplacementDrive::prepare(&params)?;
                    self.sagas.saga_start(saga_dag).await
                }.await;
```